### PR TITLE
DRFT 1179 collect all errors from parallelized api calls

### PR DESCRIPTION
**Pull request by petrblaho**
- Adds few tests for API calls from comparison_report
- Adds collecting of errors from parallel API calls

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Comment by petrblaho** [comment]
https://issues.redhat.com/browse/DRFT-1179

**Comment by petrblaho** [comment]
https://issues.redhat.com/browse/DRFT-1179

**Comment by petrblaho** [comment]
This PR should improve handling of errors during parallel API calls from comparison_report.

Originally we returned the first error that occurred in sequence of calls (inventory, baselines, hsps) thus our test suite fails with current parallel code in stage env.

This PR handles return values and codes for different errors on different API calls with precedence of 5xx, 400, 404, 2xx.

File: drift/views/v1.py [hunk 1]
```diff
@@ -285,6 +287,30 @@ def get_historical_sys_profiles(
...
+                    http_code = HTTPStatus.BAD_REQUEST
+                elif any(
+                    type(other_error) == ItemNotReturned for other_error in other_errors
+                ) or any(http_error.code == HTTPStatus.NOT_FOUND for http_error in http_errors):
+                    http_code = HTTPStatus.NOT_FOUND
+                else:
+                    http_code = HTTPStatus.INTERNAL_SERVER_ERROR
+
+                message = ", ".join(
+                    [http_error.reason for http_error in http_errors]
```
Comment (vkrizan) [inline drift/views/v1.py:310]: I'd suggest to add `str()` to be sure that the join would join strings.

File: drift/views/v1.py [hunk 2]
```diff
@@ -285,6 +287,30 @@ def get_historical_sys_profiles(
...
+                    + [other_error.message for other_error in other_errors]
+                )
+                raise HTTPError(http_code, message=message)
```
Comment (vkrizan) [inline drift/views/v1.py:313]: Can this be moved to a function? This function is already large enough.

File: drift/views/v1.py [hunk 3]
```diff
@@ -285,6 +287,30 @@ def get_historical_sys_profiles(
                             str(HTTPStatus.FORBIDDEN) + " " + message, request=request
                         )
                         raise HTTPError(HTTPStatus.FORBIDDEN, message=message)
+                    except HTTPError as error:
+                        http_errors.append(error)
+                    except (ItemNotReturned, ServiceError) as error:
+                        other_errors.append(error)
+
+            if http_errors or other_errors:
+                if any(type(other_error) == ServiceError for other_error in other_errors) or any(
```
Comment (vkrizan) [inline drift/views/v1.py:296]: `isinstance()` should be preferred to type checks.

**Review by vkrizan — APPROVED** [review]
Nice

File: drift/views/v1.py (hunk 1 above)
Comment (petrblaho) [reply drift/views/v1.py]: Good suggestion!

File: drift/views/v1.py (hunk 2 above)
Comment (petrblaho) [reply drift/views/v1.py]: Done.

File: drift/views/v1.py (hunk 3 above)
Comment (petrblaho) [reply drift/views/v1.py]: Done.

**Comment by petrblaho** [comment]
@vkrizan review comments solved. I will squash it before merging when acked.

File: drift/views/v1.py [hunk 4]
```diff
@@ -285,6 +309,14 @@ def get_historical_sys_profiles(
...
                         )
                         raise HTTPError(HTTPStatus.FORBIDDEN, message=message)
+                    except HTTPError as error:
+                        http_errors.append(error)
+                    except (ItemNotReturned, ServiceError) as error:
+                        other_errors.append(error)
+
+            final_error = merge_errors(http_errors=http_errors, other_errors=other_errors)
+            if final_error:
+                raise final_error
```
Comment (vkrizan) [inline drift/views/v1.py:319]: Raise this right in the method, then this part can be removed.

**Pull request merged**